### DISCUSSION TITLE
任务栏里的正在运行的应用如何禁止“驻留”功能

### DISCUSSION THREAD

**Discussion by DaoYangM** [linuxdeepin/developer-center#5911] [Developer Discussion | 开发者讨论]
我有一个应用A，运行的时候会在任务栏里有图标，需要禁止“驻留”这个功能，请问如何实现？

dde-dock 2.0

**Comment by BLumia** [comment]
v23 后续计划是没有对应的 .desktop 文件的应用是无法驻留的，如果你的应用符合这个情况的话应该无需任何额外处理。

不过，为什么会有这个需求？可以描述下相关场景吗？

**Reply by DaoYangM** [reply]
你好。
我现在有个程序A，程序A会打开程序B，A和B在运行的时候都要在任务栏上展示图标，但是要求B不能添加到常驻任务栏（因为，B是不能独立打开的，要求从A里启动B）。

**Reply by DaoYangM** [reply]
然后，你们是怎么识别任务栏里正在运行应用的？是通过.desktop里的Exec吗？我看设置了StartupWMClass是没用的。

**Reply by BLumia** [reply]
你是在给v20适配吗？v23的规划是必须有desktop文件对应才可以固定所以你的需求其实不需要额外写什么东西。v20的话，社区版不会有行为变化了。

**Comment by DaoYangM** [comment]
@BLumia 能麻烦回复一下吗？

**Reply by BLumia** [reply]
抱歉今天有点忙，现在才看到通知。已经在上面回复了。

顺便，我们有开发者用的实时交流群(Telegram或Matrix)，如果有兴趣的话那里提问应该得到答复会更及时一些。
